core/src/main/scala/org/apache/spark/TestUtils.scala

```diff
@@ -179,11 +179,18 @@ private[spark] object TestUtils {
       destDir: File,
       toStringValue: String = "",
       baseClass: String = null,
-      classpathUrls: Seq[URL] = Seq.empty): File = {
+      classpathUrls: Seq[URL] = Seq.empty,
+      preClassDefinitionBlock: String = "",
+      implementsClasses: Seq[String] = Seq.empty,
+      extraCodeBody: String = ""): File = {
     val extendsText = Option(baseClass).map { c => s" extends ${c}" }.getOrElse("")
+    val implementsText = implementsClasses.map(", " + _).mkString
```

Comment:
maybe?
```suggestion
    val implementsText =
      s"implements ${implementsClasses ++ Seq("java.io.Serializable").mkString(", ")}"
```

Reply:
Thanks. To be more simpler, I'll make it `(implementsClasses :+ "java.io.Serializable").mkString(", ")`.